diff --git a/libavfilter/Makefile b/libavfilter/Makefile
index 30cc329fb6..02b4c75470 100644
--- a/libavfilter/Makefile
+++ b/libavfilter/Makefile
@@ -549,6 +549,8 @@ OBJS-$(CONFIG_YAEPBLUR_FILTER)               += vf_yaepblur.o
 OBJS-$(CONFIG_ZMQ_FILTER)                    += f_zmq.o
 OBJS-$(CONFIG_ZOOMPAN_FILTER)                += vf_zoompan.o
 OBJS-$(CONFIG_ZSCALE_FILTER)                 += vf_zscale.o
+OBJS-$(CONFIG_GLTRANSITION_FILTER)			 += vf_gltransition.o
+OBJS-$(CONFIG_SHADERTOY_FILTER)			 	 += vf_shadertoy.o
 
 OBJS-$(CONFIG_ALLRGB_FILTER)                 += vsrc_testsrc.o
 OBJS-$(CONFIG_ALLYUV_FILTER)                 += vsrc_testsrc.o
diff --git a/libavfilter/allfilters.c b/libavfilter/allfilters.c
index 5ebacfde27..9053f030fb 100644
--- a/libavfilter/allfilters.c
+++ b/libavfilter/allfilters.c
@@ -516,6 +516,8 @@ extern const AVFilter ff_vf_yaepblur;
 extern const AVFilter ff_vf_zmq;
 extern const AVFilter ff_vf_zoompan;
 extern const AVFilter ff_vf_zscale;
+extern const AVFilter ff_vf_gltransition;
+extern const AVFilter ff_vf_shadertoy;
 
 extern const AVFilter ff_vsrc_allrgb;
 extern const AVFilter ff_vsrc_allyuv;
